nodeinfo/manager.go

```diff
@@ -71,43 +82,54 @@ func init() {
 }
 
 // NewInfoManager new info manager
-func NewInfoManager(cfg *Config, t transmitter, h chain, privKey crypto.PrivateKey) *InfoManager {
+func NewInfoManager(cfg *Config, t transmitter, h chain, privKey crypto.PrivateKey, fun delegatesGetFunc) *InfoManager {
 	dm := &InfoManager{
-		nodeMap:     lru.New(cfg.NodeMapSize),
-		transmitter: t,
-		chain:       h,
-		privKey:     privKey,
-		version:     version.PackageVersion,
-		address:     privKey.PublicKey().Address().String(),
-	}
-
-	dm.broadcastTask = routine.NewRecurringTask(func() {
+		nodeMap:      lru.New(cfg.NodeMapSize),
+		transmitter:  t,
+		chain:        h,
+		privKey:      privKey,
+		version:      version.PackageVersion,
+		address:      privKey.PublicKey().Address().String(),
+		getDelegates: fun,
+	}
+	dm.isDelegate.Store(false)
+	// init recurring tasks
+	broadcastTask := routine.NewRecurringTask(func() {
+		ctx := protocol.WithFeatureCtx(
+			protocol.WithBlockCtx(
+				genesis.WithGenesisContext(context.Background(), dm.chain.Genesis()),
+				protocol.BlockCtx{BlockHeight: dm.chain.TipHeight()},
+			),
+		)
+		if !protocol.MustGetFeatureCtx(ctx).EnableNodeInfo {
+			return
+		}
 		// delegates or nodes who are turned on will broadcast
-		if cfg.EnableBroadcastNodeInfo || dm.isDelegate() {
+		if cfg.EnableBroadcastNodeInfo || dm.isDelegate.Load().(bool) {
 			if err := dm.BroadcastNodeInfo(context.Background()); err != nil {
 				log.L().Error("nodeinfo manager broadcast node info failed", zap.Error(err))
 			}
 		} else {
 			log.L().Debug("nodeinfo manager general node disabled node info broadcast")
 		}
 	}, cfg.BroadcastNodeInfoInterval)
+	updateDelegateCacheTask := routine.NewRecurringTask(func() {
```

Comment:
i think can combine this part into `broadcastTask`?
```
if !dm.isDelegate.Load().(bool) {
    // update delegate
}
```
at every `BroadcastNodeInfoInterval` (= 5min now), if I'm delegate, broadcast the info, otherwise check/update whether I'm becoming a delegate

Reply:
and there's no need to add 
`DelegateCacheTTL time.Duration `yaml:"delegateCacheTTL"``